### PR TITLE
Replace and repost announced posts with invites

### LINKED ISSUE #149: Icarus nuking news posts with discord invites
With Icarus completely nuking any message with a discord invite (which includes discord events) from other community servers, it leaves a weird hole where the message itself isnt bad or goes to a bad server but because it includes an invite or event it gets removed and then the discord server notice appears. Community posts can't DM members and so this community post becomes lost. (See most recent example here: https://discord.com/channels/96335850576556032/876540480396079135/976940564454977586)

Proposal:
Because Icarus is smart enough to detect if the user posting the invite is a bot, continue to remove the offending post but then repost it without the discord link with [Discord Link] added so that the original message is preserved and so that the contents of the post can still be read. This will essentially read like this:

Original post: `Hey, I'm doing something fun in this server, come check it out discord.gg/ldsg`
Modified post: `Hey, I'm doing something fun in this server, come check it out [Discord link]`
Icarus: It is difficult to know what will be in another Discord server at any given time. If you feel that this server is appropriate to share, please only do so in direct messages.

Doing so will continue to allow external communities of which we have a playerbase (or have set up community posts to come in from external servers) to hype their various events whilst protecting our members from potentially unsafe servers.

This behavior **should not** apply when a community post contains a auto-mute word, those should continue to be deleted.

### PR DESCRIPTION
**Pull request by BobbyTheCatfish**
Resolves #149

### CODE REVIEW

File: modules/01-filter.js [hunk 1]
```diff
@@ -118,15 +118,21 @@ function processMessageLanguage(old, msg) {
...
 
     Promise.all(foundInvites).then((invites) => {
       if (invites.length > 0) {
         const external = invites.reduce((e, i) => (i && i.guild && (i.guild.id != sf.ldsg) ? e.concat(`Guild: ${i.guild.name}`, `Channel: ${i.channel.name}`) : e), ["External Discord Server Invite"]);
         if (external.length > 1) {
+          if (msg.webhookId) {
+            for (const invite of matched) msg.content = msg.content.replace(invite, "[Discord Invite]");
+            u.clean(msg, 0);
+            msg.embeds = [u.embed().setAuthor({ name: msg.author.username, iconURL: msg.author.displayAvatarURL() }).setDescription(msg.content)].concat(msg.embeds);
+            return msg.channel.send({ embeds: msg.embeds, attachments: msg.attachments });
```
Comment (LaughLax) [inline modules/01-filter.js:134]:
```suggestion
            const embeds = [u.embed({ author: msg.author }).setDescription(msg.content)].concat(msg.embeds);
            return msg.channel.send({ embeds: embeds, attachments: msg.attachments });
```

File: modules/01-filter.js [hunk 2]
```diff
@@ -138,6 +144,12 @@ function processDiscordInvites(msg) {
       }
     }).catch(e => {
       if (e && e.message == "Unknown Invite") {
+        if (msg.webhookId) {
+          for (const invite of matched) msg.content = msg.content.replace(invite, "[Discord Invite]");
+          u.clean(msg, 0);
+          msg.embeds = [u.embed().setAuthor({ name: msg.author.username, iconURL: msg.author.displayAvatarURL() }).setDescription(msg.content)].concat(msg.embeds);
+          return msg.channel.send({ embeds: msg.embeds, attachments: msg.attachments });
```
Comment (LaughLax) [inline modules/01-filter.js:151]:
```suggestion
          const embeds = [u.embed({ author: msg.author }).setDescription(msg.content)].concat(msg.embeds);
          return msg.channel.send({ embeds: embeds, attachments: msg.attachments });
```

**Review by LaughLax — CHANGES_REQUESTED** [review]
Looks good; I've suggested a couple minor changes that will keep the code a bit less verbose. Accept those and we'll get this merge going

**Comment by BobbyTheCatfish** [comment]
The changes have been committed

**Review by LaughLax — APPROVED** [review]

**Pull request merged**